app/src/main/java/com/example/taskyy/domain/navigation/Navigation.kt

```diff
@@ -107,29 +112,82 @@ fun Nav() {
             composable(route = Screen.Reminder.route + "/{dateString}") {
                 val reminderViewModel = hiltViewModel<ReminderViewModel>()
                 val state by reminderViewModel.state.collectAsState()
+                val editedRemindDescription = navController
+                    .currentBackStackEntry
+                    ?.savedStateHandle
+                    ?.getStateFlow<String>("reminderDescription", "Reminder Description")
+                    ?.collectAsStateWithLifecycle()
+
+                LaunchedEffect(editedRemindDescription) {
+                    reminderViewModel.setReminderDescription(
+                        editedRemindDescription?.value ?: "Reminder Description"
+                    )
+                }
+
                 ReminderScreen(
-                    state = state,
-                    onEvent = { event ->
-                        when (event) {
-                            is ReminderEvent.EnterReminderDescription -> navController.navigate(
-                                Screen.ReminderDetails.route
+                    state = state
+                ) { event ->
+                    when (event) {
+                        is ReminderEvent.EnterReminderDescription -> {
+                            navController.previousBackStackEntry?.savedStateHandle?.set(
+                                "screenType",
+                                event.editDescription
                             )
```

Comment:
You're here on the reminder detail screen and edit the previous back stack entry, so the one of the agenda screen. That's not correct